### PR TITLE
fix (core) Loot duplicates\restricitons

### PR DESCRIPTION
**Pull request by acidmanifesto**
## Changes Proposed:
- fix a issue dealing with duplicates and weird restrictions with looting a single creature\gameobject.  

TC Cherry pick partially: https://github.com/TrinityCore/TrinityCore/commit/ca280dbe051f2d9c43bbda128a6aa6bc19d8149f
Co-Authored-By: Shauren <shauren.trinity@gmail.com>

## Issues Addressed:

- Closes 

## Tests Performed:

- builds
- performs

## How to Test the Changes:

1. best way to test is to kill some raid\dung bosses and see if loot is still being excessive or not.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -398,6 +398,10 @@ bool Creature::InitEntry(uint32 Entry, const CreatureData* data)
             --diff;
     }
 
+    // Initialize loot duplicate count depending on raid difficulty
+    if (GetMap()->Is25ManRaid())
+        loot.maxDuplicates = 3;
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Creature/Creature.cpp:403]: brackets

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 2]
```diff
@@ -381,6 +381,10 @@ bool GameObject::Create(ObjectGuid::LowType guidlow, uint32 name_id, Map* map, u
     LastUsedScriptID = GetGOInfo()->ScriptId;
     AIM_Initialize();
 
+    // Initialize loot duplicate count depending on raid difficulty
+    if (map->Is25ManRaid())
+        loot.maxDuplicates = 3;
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/GameObject/GameObject.cpp:386]: brackets

File: src/server/game/Loot/LootMgr.cpp [hunk 3]
```diff
@@ -72,13 +72,8 @@ struct LootGroupInvalidSelector : public Acore::unary_function<LootStoreItem*, b
             uint8 foundDuplicates = 0;
             for (std::vector<LootItem>::const_iterator itr = _loot.items.begin(); itr != _loot.items.end(); ++itr)
                 if (itr->itemid == item->itemid)
-                {
-                    ++foundDuplicates;
-                    if (_proto->InventoryType == 0 && foundDuplicates == 3 && _proto->ItemId != 47242 /*Trophy of the Crusade*/) // Non-equippable items are limited to 3 drops
-                        return true;
-                    else if (_proto->InventoryType != 0 && foundDuplicates == 1) // Equippable item are limited to 1 drop
+                    if (++foundDuplicates == _loot.maxDuplicates)
                         return true;
```
Comment (Winfidonarleyan) [inline src/server/game/Loot/LootMgr.cpp:76]: brackets

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by temperrr** [comment]
Not noticing any new odd behavior or stuff but only tested with vanilla content

**Comment by acidmanifesto** [comment]
@temperrr if we are not getting the desired results as we hope we can close the pr.

**Pull request closed without merging**